tests/unit_tests/CMakeLists.txt

```diff
@@ -11,3 +11,4 @@ add_subdirectory(optimization)
 add_subdirectory(operator_tests)
 add_subdirectory(flow_variable_tests)
 add_subdirectory(ode_solver_unit_test)
+add_subdirectory(linear_solver)
```

Comment:
Would it be more appropriate to specify NNLS solver in the folder name?

Reply:
I kept is as linear solver just in case over solvers get added to the linear solver folder in the source code, and then there unit tests can be added here

Reply:
Sounds good.